perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -51,31 +51,67 @@ public Resource createResource(PerunSession sess, Resource resource, Vo vo, Faci
+			Resource existingResource = getResourcesManagerImpl().getResourceByName(sess, destinationVo, destinationFacility, destinationResource.getName());
 			throw new ResourceExistsException(existingResource);
 		} catch (ResourceNotExistsException e){
-			//if it is not taken we create new Resource in DB, then copy everything possible from template resource to our new one.
-			newResource.setName(newResourceName);
+			newResource.setName(destinationResource.getName());
 			newResource = getResourcesManagerImpl().createResource(sess, destinationVo, newResource, destinationFacility);
 
-			// We don't copy group_resource/member_resource attributes because they are different for each VO and needs to be set manually.
-			List<Attribute> templateResourceAttributes = perunBl.getAttributesManagerBl().getAttributes(sess,resource);
+			//resource attributes
+			List<Attribute> templateResourceAttributes = perunBl.getAttributesManagerBl().getAttributes(sess,templateResource);
 			for (Attribute resourceAttribute : templateResourceAttributes) {
 				try {
-					if (!resourceAttribute.getNamespace().startsWith(AttributesManager.NS_RESOURCE_ATTR_VIRT)) {
+					if (!resourceAttribute.getNamespace().startsWith(AttributesManager.NS_RESOURCE_ATTR_VIRT) &&
+							!resourceAttribute.getNamespace().startsWith(AttributesManager.NS_RESOURCE_ATTR_CORE)) {
 						perunBl.getAttributesManagerBl().setAttribute(sess, newResource, resourceAttribute);
 					}
 				} catch (WrongAttributeValueException | WrongAttributeAssignmentException | WrongReferenceAttributeValueException ex) {
 					throw new ConsistencyErrorException("DB inconsistency while copying attributes from one resource to another. Cause:{}", ex);
 				}
 			}
 
+			//if the VOs are same we also copy groups and group-resource/member-resource attributes
+			if(withGroups && destinationVo.equals(this.getVo(sess, templateResource))){
+				List<Group> templateResourceGroups = perunBl.getResourcesManagerBl().getAssignedGroups(sess, templateResource);
+				try {
+					assignGroupsToResource(sess, templateResourceGroups, newResource);
+					for (Group group : templateResourceGroups) {
+						List<Attribute> groupResourceAttrs = perunBl.getAttributesManagerBl().getAttributes(sess, templateResource, group);
+						for (Attribute attr : groupResourceAttrs) {
+							if (!attr.getNamespace().startsWith(AttributesManager.NS_GROUP_RESOURCE_ATTR_VIRT)) {
+								perunBl.getAttributesManagerBl().setAttribute(sess, newResource, group, attr);
+							}
+						}
+					}
+				} catch (GroupResourceMismatchException | WrongAttributeValueException | GroupAlreadyAssignedException |
+					WrongAttributeAssignmentException | WrongReferenceAttributeValueException ex) {
+					throw new ConsistencyErrorException("DB inconsistency while copying group-resource attributes. Cause:{}", ex);
+				}
+
+				List<Member> templateResourceMembers = perunBl.getResourcesManagerBl().getAssignedMembers(sess, templateResource);
+				try {
```

Comment:
I'm not sure if this try/catch block is correct because we are assigning member-resource attributes to non-existing members in new Resource. Should i copy all members from template resource to the new one (same idea as with groups above)? If so, should i assign them into corresponding groups?

Reply:
It is little complicated. Members are always part of existing group. If you are copying groups from resource A to resource B, it means that all members on resource A will be also members on resource B. So for those members you can set member-resource attributes.